infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -91,17 +98,81 @@ public LogReplicationConfig(Set<String> streamsToReplicate, int maxNumMsgPerBatc
+        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, MAX_CACHE_NUM_ENTRIES, null);
     }
 
-    public LogReplicationConfig(Set<String> streamsToReplicate, Map<UUID, List<UUID>> streamingTagsMap,
-                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize, int cacheSize) {
-        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize);
+    public LogReplicationConfig(Set<TableInfo> streamsToReplicate, Map<UUID, List<UUID>> streamingTagsMap,
+                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize,
+                                int cacheSize, LogReplicationStreamNameTableManager streamInfoManager) {
+        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize, streamInfoManager);
         this.dataStreamToTagsMap = streamingTagsMap;
         this.mergeOnlyStreams = mergeOnlyStreams;
     }
-}
\ No newline at end of file
+
+    /**
+     * This class represents Log Replication streams info
+     */
+    public static class StreamInfo {
+        private final Set<UUID> streamIds;
+        private final LogReplicationStreamNameTableManager streamInfoManager;
+
+        public StreamInfo(@NonNull Set<TableInfo> streamsToReplicate,
+                          LogReplicationStreamNameTableManager streamInfoManager) {
+            this.streamInfoManager = streamInfoManager;
+            this.streamIds = new HashSet<>();
+            refreshStreamIds(streamsToReplicate, true);
+        }
+
+        private void refreshStreamIds(Set<TableInfo> streamsToReplicate, boolean shouldClear) {
+            if (shouldClear) {
+                this.streamIds.clear();
+            }
+
+            for (TableInfo info : streamsToReplicate) {
+                if (info.hasField(TableInfo.getDescriptor().findFieldByName("name"))) {
+                    this.streamIds.add(CorfuRuntime.getStreamID(info.getName()));
+                } else if (info.hasField(TableInfo.getDescriptor().findFieldByName("id"))) {
+                    this.streamIds.add(UUID.fromString(info.getId()));
+                }
+            }
+        }
+
+        public void syncWithInfoTable() {
+            if (streamInfoManager == null) {
+                log.warn("streamInfoManager is null! skipping sync!");
+                return;
+            }
+
+            Set<TableInfo> fetched = this.streamInfoManager.getStreamsToReplicate();
+            refreshStreamIds(fetched, true);
+        }
+
+        public void syncWithTableRegistry(long timestamp) {
+            if (streamInfoManager == null) {
+                log.warn("streamInfoManager is null! skipping sync!");
+                return;
+            }
+
+            Set<TableInfo> registryInfo = streamInfoManager.readStreamsToReplicatedFromRegistry(timestamp);
+            refreshStreamIds(registryInfo, true);
+        }
+
+        public Set<UUID> getStreamIds() {
+            return ImmutableSet.copyOf(streamIds);
```

Comment:
this would throw an NPE if stream ids is null

Reply:
I thought the `streamsIds` set is guaranteed to be non-null as per its data flow, but it could be empty.